dmn/src/main/java/io/camunda/zeebe/dmn/impl/DmnScalaDecisionEngine.java

```diff
@@ -106,8 +106,15 @@ public DecisionEvaluationResult evaluateDecisionById(
 
     if (result.isLeft()) {
       final var reason = result.left().get().failure().message();
+
+      String failedDecisionId = null;
+      if (!evaluatedDecisions.isEmpty()) {
+        failedDecisionId = evaluatedDecisions.get(evaluatedDecisions.size() - 1).decisionId();
+      }
```

Comment:
:thought_balloon: In Scala, we could write:

```
val failedDecisionId = evaluatedDecisions.lastOption.orNull
``` 

I don't see an easy functional way in Java :disappointed:

Reply:
Not as pretty, but:
```java
evaluatedDecisions.stream()
    .reduce((a, b) -> b)
    .map(EvaluatedDecision::decisionId)
    .orElse(null)
```

Reply:
You could also consider a util method, like suggested [here](https://stackoverflow.com/a/55593682)